TWNIA2/2. Tutorials/3. 使用 Conda 管理套件與執行 Job.md

```diff
@@ -1 +1,229 @@
+tensorflow-gpu version 2.2 is currently installed by default
+- You can choose to install version 1 or 2, for example, use 
+ `conda create --name mytf_gpu tensorflow-gpu=1.15.0` for 1.15.0 version of tensorflow-gpu
+ - For other versions, refer to [<ins>version list</ins>](https://anaconda.org/anaconda/tensorflow-gpu/files).
+:::
+
+#### Deactivate and delete the virtual environment
+```bash
+# deactivate virtual environment named mytf_gpu
+$ conda deactivate
+
+# delete virtual environment named mytf_gpu 
+$ conda remove --name mytf_gpu --all
+```
+
+#### Unload Conda
+```bash
+# Uninstall all loaded modules
+$ module purge
+```
+
+```bash
+# Or, only unload miniconda
+$ module unload miniconda3
+```
+
+#### Delete remaining Conda files
+```bash
+# When Conda is no longer used, delete the remaining Conda installation files/configuration files (please use it carefully!! Make sure there is no need for files in this path before using this command)
+# Check the contents of this path file first (usually only ~/.conda)
+$ ls ~/.condarc ~/.conda ~/.continuum
+
+# Then, delete the files
+$ rm -rf ~/.condarc ~/.conda ~/.continuum
+```
+
+### Step 2. Slurm job script with Conda
+
+Once you have prepared the Conda environment and computing program, please follow the steps to write a job script using Slurm Workload Manager to request resources, queue schedules, and submit jobs according to your needs.
+
+
+The `.sh` file is the job script submitted by Slurm, the content is divided into two parts:
+
+1. **Information of Job, project and resources**: job name, Number of nodes, Number of jobs running per node, Number of GPUs per node, longest job running time, project ID, and queue name.
+3. **The content of the job**
+
+
+Write the following content and use the virtual environment `mytf_gpu` created by the above steps.
+
+
+Run the following edit command > Press <kbd>i</kbd> key > Copy and paste the example > Press <kbd>ESC</kbd> key > Enter `:wq!` to save and exit. We're done!
+
+
+```bash
+$ vim <FILE_NAME>.sh
+```
+
+:::info 
+<i class="fa fa-paperclip fa-20" aria-hidden="true"></i> **Note:** You may write `.sh`  files using the editor you're used to, the example uses vim to operate.
+:::
+
+- TensorFlow 2 with 1 node / 8GPUs
+```bash
+#!/bin/bash
+#SBATCH --job-name=Hello_TWCC    ## job name
+#SBATCH --nodes=1                ## request 1 node
+#SBATCH --cpus-per-task=32       ## request 32 CPUs for that task
+#SBATCH --gres=gpu:8             ## request 8 GPUs per node
+#SBATCH --time=00:10:00          ## Run for up to 10 minutes (remember to change it after testing, or delete this line directly)
+#SBATCH --account=PROJECT_ID     ## PROJECT_ID (e.g.,  MST108XXX). Billing will also be based on this project ID
+#SBATCH --partition=gtest        ## gtest is used for testing queue. After subsequent tests, it can be changed to gp1d (maximum running 1 day), gp2d (maximum running 2 days), gp4d (maximum running 4 days)
```

Comment:
gtest is used for testing queue. After subsequent tests, it can be changed to gp1d (maximum running 1 day), gp2d (maximum running 2 days), gp4d (maximum running 4 days) >

gtest is a testing queue. After testing, you can change the queue to  gp1d (Max. time: 1 day), gp2d (Max. time: 2 days), or gp4d (Max. time: 4 days)